api/nodes/serializers.py

```diff
@@ -640,3 +644,94 @@ def error_checker(self, data):
     def get_absolute_url(self, obj):
         #  Citations don't have urls
         raise NotImplementedError
+
+
+class DraftRegistrationSerializer(JSONAPISerializer):
+
+    id = IDField(source='_id', read_only=True)
+    type = TypeField()
+    registration_supplement = ser.CharField(source='registration_schema._id', required=True)
+    registration_metadata = ser.DictField(required=False)
+    datetime_initiated = ser.DateTimeField(read_only=True)
+    datetime_updated = ser.DateTimeField(read_only=True)
+
+    branched_from = RelationshipField(
+        related_view='nodes:node-detail',
+        related_view_kwargs={'node_id': '<branched_from._id>'}
+    )
+
+    initiator = RelationshipField(
+        related_view='users:user-detail',
+        related_view_kwargs={'user_id': '<initiator._id>'},
+    )
+
+    registration_schema = RelationshipField(
+        related_view='metaschemas:metaschema-detail',
+        related_view_kwargs={'metaschema_id': '<registration_schema._id>'}
+    )
+
+    links = LinksField({
+        'html': 'get_absolute_url'
+    })
+
+    def get_absolute_url(self, obj):
+        return obj.absolute_url
+
+    def create(self, validated_data):
+        node = validated_data.pop('node')
+        initiator = validated_data.pop('initiator')
+        metadata = validated_data.pop('registration_metadata', None)
+
+        schema_id = validated_data.pop('registration_schema').get('_id')
+        schema = get_object_or_error(MetaSchema, schema_id)
+        if schema.schema_version != 2 or schema.name not in ACTIVE_META_SCHEMAS:
+            raise exceptions.ValidationError('Registration supplement must be an active schema.')
+
+        draft = DraftRegistration.create_from_node(node=node, user=initiator, schema=schema)
+        if metadata:
+            self.validate_metadata(draft, metadata)
+            draft.update_metadata(metadata)
+            draft.save()
+        return draft
+
+    def validate_metadata(self, draft, metadata):
+        """
+        Validates registration_metadata field.  Called in update and create methods because the draft is
+        needed in the context.
+        """
+        reviewer = is_prereg_admin_not_project_admin(self.context['request'], draft)
+        # Required fields are only required when creating the actual registration, not updating the draft.
+        schema = create_jsonschema_from_metaschema(draft, required_fields=False, is_reviewer=reviewer)
+        try:
+            jsonschema.validate(metadata, schema)
```

Comment:
This is pretty neat. It might be good to move this validation logic to the model, so we can hide these implementation details. The API would be something like

```python
draft.validate(metadata, ....)
```